### ISSUE TITLE
TripletSemiHardLoss produces only NaNs when training with a TPU

### ISSUE THREAD

**Issue by kbelenky** [tensorflow/tensorflow#53567]
This bug was discovered in Google Colab, but appears to be rooted in TripletSemiHardLoss.

Here's an example Colab that illustrates the problem:
https://colab.research.google.com/drive/10GuYAFDMuxXe2jqdj7hy_I7XcApJFmj7?usp=sharing

- Have I written custom code (as opposed to using a stock example script provided in TensorFlow):

I combined the example Colab here:
https://colab.research.google.com/github/GoogleCloudPlatform/training-data-analyst/blob/master/courses/fast-and-lean-data-science/01_MNIST_TPU_Keras.ipynb

with some of the steps here:
https://www.tensorflow.org/addons/tutorials/losses_triplet

Specifically:
I deleted the line: `label = tf.one_hot(label, 10)`
I replaced the last layer of the network (Dense ReLU) with these two layers:
```
 tf.keras.layers.Dense(10, activation=None),
 tf.keras.layers.Lambda(lambda x: tf.math.l2_normalize(x, axis=1)) # L2 normalize embeddings
```

And then changed the loss function in `make_model` to: `tfa.losses.TripletSemiHardLoss(distance_metric='angular', margin=0.2)`

When run on a CPU or GPU, the training works as expected and the loss function produces reasonable results.
When run on a TPU, the training runs (as in it does not produce an error), but the loss function produces only NaN, causing the training to effectively fail.

I understand that not all operations are available on TPUs and TripletSemiHardLoss may fundamentally not work on a TPU, but it should at least fail with an informative error message.

**Comment by sushreebarsa** [comment]
@kbelenky 
Please post this issue on [keras-team/keras repo.](https://github.com/keras-team/keras/issues)
To know more see;
[https://discuss.tensorflow.org/t/keras-project-moved-to-new-repository-in-https-github-com-keras-team-keras/1999](https://discuss.tensorflow.org/t/keras-project-moved-to-new-repository-in-https-github-com-keras-team-keras/1999)
Thank you!
